jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -256,16 +257,12 @@ private static InsnNode simplifyArith(ArithNode arith) {
 			}
 			InsnArg firstArg = arith.getArg(0);
 			if (arith.getOp() == ArithOp.XOR && firstArg.getType() == ArgType.BOOLEAN) {
-				if (lit == 1 && firstArg.isInsnWrap()) {
-					InsnNode wr = ((InsnWrapArg) firstArg).getWrapInsn();
-					InsnNode not = new InsnNode(InsnType.NOT, 1);
-					not.setResult(wr.getResult());
-					not.addArg(firstArg);
-					return not; 
-				}
-				if (lit == 0 && firstArg.isInsnWrap()) {
-					return ((InsnWrapArg) firstArg).getWrapInsn();
-				}
+				InsnNode node = new InsnNode(lit == 0 ? InsnType.MOVE : InsnType.NOT, 1);
```

Comment:
Please ignore cases other then `lit == 0` and `lit == 1`.
I know such cases seem impossible, but better treat them well and don't produce wrong code :)